### ISSUE TITLE
Builds will Run Multiple Times for the Same Artifact

### ISSUE THREAD

**Issue by efrisch** [pyrsia/pyrsia#1030]
### Steps taken/Steps to reproduce

`./target/debug/pyrsia build maven --gav commons-codec:commons-codec:1.15`
then a second
`./target/debug/pyrsia build maven --gav commons-codec:commons-codec:1.15`

And it will happily accept a second build of the same thing.

### Expected Behavior

Should not build the same artifact a second time.

### Experienced Instance of the bug

Rust: rustc 1.63.0 (4b91a6ea7 2022-08-08)
Operating System: mac 12.5.1
Pyrsia version: Built off `main` (pre 0.2)

**Comment by jperedadnr** [comment]
With current code, it is still the case:
- nodeA (authorized)
- pyrsia points to nodeB
- launching almost from terminal one:
```
./pyrsia build docker --image alpine:3.16.2
Build request successfully handled. Build with ID 1 has been started.
```
and almost at the same time from terminal two:
```
./pyrsia build docker --image alpine:3.16.2
Build request successfully handled. Build with ID 2 has been started.
```

Builds twice and produces 8 artifacts in node A (4 per build).

nodeB logs:

```
// build 1
pyrsia::artifact_service::service    > Request build of Docker "alpine:3.16.2"
pyrsia::network::build_protocol      > Write BuildRequest: Docker: alpine:3.16.2
pyrsia::network::event_loop          > RequestResponseMessage::Response RequestId(1)
pyrsia_registry                      > 127.0.0.1:52365 "POST /build/docker HTTP/1.1" 200 "-" "-" 9.9045ms
// build 2
pyrsia::artifact_service::service    > Request build of Docker "alpine:3.16.2"
pyrsia::network::build_protocol      > Write BuildRequest: Docker: alpine:3.16.2
pyrsia::network::event_loop          > RequestResponseMessage::Response RequestId(2)
pyrsia_registry                      > 127.0.0.1:52372 "POST /build/docker HTTP/1.1" 200 "-" "-" 4.351583ms
...
// build 1:
pyrsia::transparency_log::log        > Transparency log inserted into database with id: 9a712dde-6285-4d6f-9f93-bcd7e5a2ecb8
pyrsia::transparency_log::log        > Transparency log inserted into database with id: 51d2d4fb-fdeb-4db3-b91f-b82eeae32232
pyrsia::transparency_log::log        > Transparency log inserted into database with id: 8924d813-cac9-4a10-a64b-dd0d3916de47
pyrsia::transparency_log::log        > Transparency log inserted into database with id: 43fc98a9-4db7-40ff-9ede-86b554f474b1
// build 2:
pyrsia::transparency_log::log        > Transparency log inserted into database with id: 080a08e2-da71-404d-84e8-59b49a9d020c
pyrsia::transparency_log::log        > Transparency log inserted into database with id: b58a62b4-4e96-4b56-9ddd-57e3a382fecc
pyrsia::transparency_log::log        > Transparency log inserted into database with id: aece49d2-4156-48ae-bc9c-0595ebc18fca
pyrsia::transparency_log::log        > Transparency log inserted into database with id: 7397c6db-e0d7-46f1-8e7b-81c5abf51642
```

**Comment by erwin1** [comment]
this was fixed in #1199
